tests/Utils.ts

```diff
@@ -75,32 +75,52 @@ export class TestAgora
         this.agora.get("/block_height",
             (req: express.Request, res: express.Response) =>
         {
-            res.status(200).send(JSON.stringify(Number(this.blocks.length - 1)));
+            res.json(new Height(BigInt(this.blocks.length - 1)));
         });
 
         this.agora.get("/blocks_from",
             (req: express.Request, res: express.Response) =>
         {
-            if  (
-                    (req.query.block_height === undefined) ||
-                    (req.query.max_blocks === undefined) ||
-                    Number.isNaN(req.query.block_height) ||
-                    Number.isNaN(req.query.max_blocks)
-                )
+            if (req.query.block_height === undefined || Number.isNaN(req.query.block_height))
+            {
+                res.status(400).json({ errorMessage: 'Missing or invalid block_height query parameter' });
+                return;
+            }
+            if (req.query.max_blocks === undefined || Number.isNaN(req.query.max_blocks))
+            {
+                res.status(400).json({ errorMessage: 'Missing or invalid max_blocks query parameter' });
+                return;
+            }
+
+            let block_height = new Height(BigInt(req.query.block_height));
+            if (block_height.value < 0n)
             {
-                res.status(200).send(JSON.stringify([]));
+                res.status(400).json({ errorMessage: 'Query parameter block_height must not be negative' });
                 return;
             }
 
-            let block_height = Math.max(Number(req.query.block_height), 0);
-            let max_blocks = Math.max(Number(req.query.max_blocks), 0);
+            // Slightly stricter requirement than Agora (strictly positive instead of positive)
+            // as we don't want Stoa to request 0 blocks
+            let max_blocks = Number(req.query.max_blocks);
+            if (max_blocks <= 0)
+            {
+                res.status(400).json({ errorMessage: 'Query parameter max_block must be strictly positive' });
+                return;
+            }
+
+            if (this.blocks.length < 1)
+            {
+                res.status(500).json({ errorMessage: 'No data on server' });
```

Comment:
`errorMessage:`
Now should we apply all response error messages key in common?

Reply:
No, I should have used `statusText`

Reply:
A `statusMessage` is used in Agora.
https://github.com/bpfkorea/vibe.d/blob/master/web/vibe/web/rest.d#L1466